### ISSUE TITLE
Calling additional columns into custom template

### ISSUE THREAD

**Issue by tyokota** [MinnPost/legislature-tracker#38]
Would like to call additional columns into custom template using something like <%= bill.x %> to retrieve data in the "x" column of the bill tab in the Google spreadsheet.

**Comment by zzolo** [comment]
I still haven't actually looked into trying this, but as you pointed out the NY World instance is doing something similar.

As it's all client side code, you can dig into it yourself.  Looking at their code and spreadsheet.
https://spreadsheets.google.com/spreadsheet/pub?key=0AgIwhjKwGPrydGs3b0taR2JTeGkyZ01kbE9rRVVBM2c&gid=1

It looks like the missing piece is that you have to explicitly tell the app to translate your field.  You can see the default options here:
https://github.com/MinnPost/legislature-tracker/blob/master/js/app/core.js#L306

Unfortunately it looks like this configuration option never made it into the README (or got removed at some point).  You'll probably want to add it a way similar to the translation options.
https://github.com/MinnPost/legislature-tracker#translation-options

And of course, you will have to add the field to spreadsheet, and override the template to add in the data.

**Comment by tyokota** [comment]
Hi Alan,

Thanks for the tips. Do I need to compile the core.js folder? If so, how? Or do I just overwrite the core.js file with the new translation options? 

I've mimicked all that I thought was needed on the client side of the code, but it appear there's something I am missing... 

Thanks,
Thomas

**Comment by zzolo** [comment]
No, fieldTranslations is a configuration option you can overwrite when you call the application.  You may want to extend the default options so that you don't have to recreate that whole fieldTranslations object (hence the link to the translation options sections).

**Comment by tyokota** [comment]
Thanks Alan. That makes sense. I'll give it a shot.

**Comment by tyokota** [comment]
I am copying this piece into the index file and extending it right? Or can this be truncated?

  // Handle changing field names
  LT.parse.translateFields = function(translation, row) {
    _.each(translation, function(input, output) {
      row[output] = row[input];

```
  if (output !== input) {
    delete row[input];
  }
});

return row;
```

  };

  // Default options
  LT.defaultOptions = {
    sheetsWanted: ['Categories', 'Bills', 'Events'],
    fieldTranslations: {
      eCategories: {
        'id': 'categoryid',
        'short_title': 'shorttitle'
      },
      eBills: {
        'bill': 'bill',
        'bill_companion': 'companionbill',
        'bill_conference': 'conferencebill',
        'categories': 'categories',
        'title': 'title',
        'description': 'description'
      },
      eEvents: {
        'bill_id': 'bill',
        'actor': 'chamber',
        'action': 'title'
      }
    },
    wordTranslations: {
      chamber: {
        'upper': 'Senate',
        'lower': 'House'
      },
      partyAbbr: {
        'Democratic-Farmer-Labor': 'DFL',
        'Democratic': 'D',
        'Republican': 'R'
      }
    },
    maxBills: 30,
    substituteMatch: (/substituted/i),
    imagePath: './css/images/',
    templatePath: './js/app/templates/',
    recentChangeThreshold: 7,
    tabletopOptions: {},
    scrollOffset: false,
    conferenceBill: true,
    recentImage: 'RecentUpdatedBill.png',
    chamberLabel: false,
    detectCompanionBill: (/([A-Z]+ [1-9][0-9]_)$/),
    billNumberFormat: (/[A-Z]+ [1-9][0-9]_/),
    osBillParse: false
  };

**Comment by zzolo** [comment]
It's just a configuration option.

```
var options = _(LT.defaultOptions).extend({
  el: '#legislature-tracker-container',
  state: 'NY', 
  session: '2013-2014',
  OSKey: 'abc',
  eKey: 'abc'
});
options['fieldTranslations']['eBills']['your_custom_field'] = 'your_custom_field';

var app = new LT.Application(options);
```

**Comment by tyokota** [comment]
I was able to input that into my options, but calling it in my template as
<% = custom_field %> did not work. Any suggestions?

**Comment by zzolo** [comment]
Hey Thomas, do you have your code anywhere that I can view it?

**Comment by tyokota** [comment]
Hi Alan,

Here is a zip of the code in various files. I hope this helps.

Thanks,
Thomas

**Comment by zzolo** [comment]
Thomas,

I wasn't able to get your code to work, and it looks like you altered the Leg Tracker code.  But, I was able to get this working on the examples.

As suspected, you can just add a field and override the template and include it and it should be there as `bill.customfield`.  A thing to note is that when Tabletop reads in column names, it removes underscores, and maybe capitalization.

Spreadsheet:
https://docs.google.com/a/minnpost.com/spreadsheet/ccc?key=0AjYft7IGrHzNdENILV9DU056SkxNQ2tIaHNGMnRlZUE#gid=1

Example (see right above the tile):
http://code.minnpost.com/legislature-tracker/examples/test-dev.html#/bill/A%206357

Code:
https://github.com/MinnPost/legislature-tracker/blob/master/examples/test-dev.html

Specifically, adding in new code to template:
https://github.com/MinnPost/legislature-tracker/blob/master/examples/test-dev.html#L38

And then overwriting template:
https://github.com/MinnPost/legislature-tracker/blob/master/examples/test-dev.html#L208

A note about Tabletop and translations:
https://github.com/MinnPost/legislature-tracker/blob/master/examples/test-dev.html#L202

In the code you sent, it looks like you want to add a field to the eBills template, but your are not overriding that.

Hope that helps.  Closing for now, but feel free to re-open if you have more questions.

**Comment by tyokota** [comment]
The example worked well and I was able to adapt it. For the sake of consistency, I would like to move the custom_field into osbill so that the code is pulled for both primary and complementary bills. I have created the custom template in index.html, and I got the application to load the new template. However, when using the provided code, osbill doesn't load the custom_field. Is there something I need to do so that I can call the custom_field from osbill template?
